modules/homa-lite/src/lib.rs

```diff
@@ -821,20 +916,26 @@ pub mod module {
 					actual_staking_amount,
 					actual_liquid_amount,
 				));
+				num_matched += 1u64;
+
+				// If all the currencies are minted, return.
+				if available_staking_balance.is_zero() || num_matched >= max_num_matches {
```

Comment:
to better handle rounding error, could stop when available_staking_balance <= minimal_balance

Reply:
This also resolves #1545 
Added test to test the bug is fixed